src/spectrum/group.rs

```diff
@@ -692,47 +692,52 @@ mod mzsignal_impl {
         }
     }
 
-    /// Average a series of [`MultiLayerSpectrum`] together
-    pub fn average_spectra<C: CentroidLike + Default, D: DeconvolutedCentroidLike + Default>(
-        spectra: &[MultiLayerSpectrum<C, D>],
+    /// Average a series of [`SpectrumLike`] together. The supplied dx will be used to [`reprofile`]
+    /// centroid spectra as well as be the dx used for any profile spectra. The resulting
+    /// [`ArrayPair`] can be made into a [`BinaryArrayMap`] using `.into()`. Which in turn can be
+    /// used to create a new [`MultiLayerSpectrum`] or [`RawSpectrum`](crate::spectrum::RawSpectrum).
+    ///
+    /// Note: only available with feature `mzsignal`.
+    ///
+    /// # Warns
+    /// Warns you when the spectrum has `SignalContinuity::Profile` but not raw data array or when
+    /// the raw data arrays differ in dx with the supplied value.
+    pub fn average_spectra<S: SpectrumLike<C, D>, C: CentroidLike, D: DeconvolutedCentroidLike>(
+        spectra: impl IntoIterator<Item = S>,
```

Comment:
To verify: does this still accept slices since there is no `impl SpectrumLike for &S where S: SpectrumLike` declaration?

Reply:
It did not so I moved to an iterator over references, this makes it somewhat less ergonomic when you have an owned iterator, but it should be easy enough to convert (easier than the other way around).